lldb/test/Shell/BoundsSafety/boundssafety_soft_trap_call_with_str.test

```diff
@@ -0,0 +1,22 @@
+# UNSUPPORTED: system-windows
+# RUN: %clang_host -fbounds-safety -fbounds-safety-soft-traps=call_with_str -g -O0 %S/Inputs/boundsSafetySoftTraps.c -o %t.out
+# RUN: %lldb -b -s %s %t.out | FileCheck %s
+b __bounds_safety_soft_trap_s
+run
+
+# TODO: We should probably teach LLDB to recognize soft traps and set the stop
+# reason appropriately
+# CHECK: * thread #{{.*}} stop reason = breakpoint 1.1
+
+# Check that reason for bounds check failing can be seen in the stacktrace
+bt
+# CHECK: * frame #{{.*}}`__bounds_safety_soft_trap_s(reason="indexing above upper bound in 'array[index]'")
+# CHECK-NEXT: frame #{{.*}}`__clang_trap_msg$Bounds check failed$indexing above upper bound in 'array[index]'
```

Comment:
So with soft-traps we still want the fake frame in debug-info because the `reason` parameter to `__bounds_safety_soft_trap_s` isn't guaranteed to be preserved on higher optimisation levels? But yea, we probably do want to make the frame recogniser aware of the soft-trap. (Probably just need to add it to the regex so the frame recogniser starts walking the backtrace).

Reply:
@Michael137 Regarding the frame recognizer. I've been talking with @jimingham and we are considering adding an `InstrumentationRuntime` plug-in that automatically sets the breakpoints on these two symbols and also sets the stop reason appropriately. If we that do still need to change the frame recognizer too?